docs/content/howto/dataframe-api.md

```diff
@@ -0,0 +1,217 @@
+
+```python
+# resample the first second of data at every millisecond
+view = view.using_index_values(range(0, 1_000_000, 1_000_0000_000))
+```
+
+In this case, the view will return rows in multiples of 1e6 nanoseconds (i.e. for each millisecond) over a period of one second. A precise match on the index value is required for data to be produced on the row. For this reason, a floating point second API is again not provided for this feature.
+
+Note that this feature is typically used in conjunction with `fill_latest_at()` (see next paragraph) to enable arbitrary resampling of the original data.
+
+
+### Filling empty values with latest-at data
+
+By default, the rows returned by the view may be sparse and contain values only for the columns where a logging event actually occurred at the corresponding index value. The view can optionally replace these empty cells using a latest-at query. This means that, for each such empty cell, the view traces back to find the last logged value and uses it instead. This is enabled by calling the `fill_latest_at()` method:
+
+```python
+view = view.fill_latest_at()
+```
+
+### Reading the data
+
+Once the view is fully set up (possibly using the filtering features previously described), its content can be read using the `select()` method. This method optionally allows specifying which subset of columns should be produced:
+
+
+```python
+# select all columns
+record_batches = view.select()
+
+# select only the specified columns
+record_batches = view.select(
+    [
+        rr.dataframe.IndexColumnSelector("frame_nr"),
+        rr.dataframe.ComponentColumnSelector("/world/robot", "Position3D"),
+    ],
+)
+```
+
+The `select()` method returns a [`pyarrow.RecordBatchReader`](https://arrow.apache.org/docs/python/generated/pyarrow.RecordBatchReader.html), which is essentially an iterator over a stream of [`pyarrow.RecordBatch`](https://arrow.apache.org/docs/python/generated/pyarrow.RecordBatch.html#pyarrow-recordbatch)es containing the actual data. See the [PyArrow documentation](https://arrow.apache.org/docs/python/index.html) for more information.
+
+In the rest of this page, we explore how these `RecordBatch`es can be ingested in some of the popular data science packages.
+
+
+## Load data to a PyArrow `Table`
+
+The `RecordBatchReader` provides a [`read_all()`](https://arrow.apache.org/docs/python/generated/pyarrow.RecordBatchReader.html#pyarrow.RecordBatchReader.read_all) method which directly produces a [`pyarrow.Table`](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table):
+
+```python
+import rerun as rr
+
+recording = rr.dataframe.load_recording("/path/to/file.rrd")
+view = recording.view(index="frame_nr", contents="/**")
+
+table = view.select().read_all()
+```
+
+
+## Load data to a Pandas dataframe <!-- NOLINT -->
```

Comment:
Instead of `NOLINT`, you can add `Pandas` and `Polars` to https://github.com/rerun-io/rerun/blob/017ef53d905c4c456a3104c3b23d4f835c103dac/scripts/lint.py#L666